splash/browser_tab.py

```diff
@@ -1078,6 +1093,65 @@ def log(self, message, min_level=None):
         log.msg(message, system='render')
 
 
+class ElementsStorage(QObject):
+    def __init__(self, parent):
+        self.name = str(uuid.uuid1())
+        super(ElementsStorage, self).__init__(parent)
+
+    @pyqtSlot(name="getId", result=str)
+    def get_id(self):
+        return str(uuid.uuid1())
+
+
+class HTMLElement(object):
+    DIMENSIONS_JS_FUNC = """
+        (function(element) {
+            var rect = element.getBoundingClientRect();
```

Comment:
True, but I still think that for getting "dimensions" the bounding variant is better. For clicking or interacting with it somehow, probably not, but this method is used later on for capturing the elements and like I said, the first client rect is most certainly not what you'd expect when you'd do `element.png()`.